### DISCUSSION TITLE
Is there any way to customize or add logic to the confirm on exit dialog in kitty?

### DISCUSSION THREAD

**Discussion by embeddedpenguin** [kovidgoyal/kitty#7855] [Q&A] [answered]
I'd like for the prompt to ask me if I'm sure I'd like to exit unless the only program running is tmux. Is this currently possible?

**Comment by kovidgoyal** [comment, marked as answer]
No, it isnt.

On Sat, Sep 14, 2024 at 11:15:18AM -0700, Penguin wrote:
> I'd like for the prompt to ask me if I'm sure I'd like to exit unless the only program running is tmux. Is this currently possible?
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/7855
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>

--
_____________________________________

Dr. Kovid Goyal
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by embeddedpenguin** [comment]
Would you be open to a patch implementing an option for a list of programs to ignore when confirming a window close? If so, how does the feature currently work with respect to jobs started in the terminal? Does it just use the job in the foreground of the shell? I guess why I'm asking is it already has to exclude the active shell the window contains.

**Comment by embeddedpenguin** [comment]
```
                    program = os.path.basename((w.child.foreground_cmdline or ('',))[0])
                    if w is active_window:
                        active_program = program
```
Looks like it only cares about whats in the fg, so implementing this should be as easy as adding a check here against the active program, x3 for window close, tab close, and quit

**Reply by embeddedpenguin** [reply]
Actually, I think this is a poor solution. Going to think about it and come back with a proposal.

**Comment by kovidgoyal** [comment]
Oh and I should note that while you cannot configure the builtin confirm_on_exit action, you can write a custom kitten that implements an exit action that does whatever you like. https://sw.kovidgoyal.net/kitty/kittens/custom/

**Reply by embeddedpenguin** [reply]
> Oh and I should note that while you cannot configure the builtin confirm_on_exit action, you can write a custom kitten that implements an exit action that does whatever you like. https://sw.kovidgoyal.net/kitty/kittens/custom/

Hey that's what my proposal was going to be :laughing: Thanks!
